### ISSUE TITLE
RUF052 false negatives for `class`, `def`, and `import` bindings

### ISSUE THREAD

**Issue by dscorbett** [astral-sh/ruff#26951]
### Summary

[`used-dummy-variable` (RUF052)](https://docs.astral.sh/ruff/rules/used-dummy-variable/) has false negatives for local variables bound by `class`, `def`, `import`–`as`, or `from`–`import`–`as`. [Example](https://play.ruff.rs/8c2346dd-50ec-4412-91c1-386041616abf):
```console
$ cat >ruf052.py <<'# EOF'
def f():
    class _C: pass
    _C
    def _f(): pass
    _f
    import rlcompleter as _rlcompleter
    _rlcompleter
    from sys import getandroidapilevel as _getandroidapilevel
    _getandroidapilevel
# EOF

$ ruff --isolated check --select RUF052 --preview ruf052.py
All checks passed!
```

### Version

ruff 0.15.22 (0177a7e0d 2026-07-16)
